pkl-cli/pkl-cli.gradle.kts

```diff
@@ -218,39 +215,48 @@ fun Exec.configureExecutable(
  * Builds the pkl CLI for macOS/amd64.
  */
 val macExecutableAmd64: TaskProvider<Exec> by tasks.registering(Exec::class) {
-  dependsOn(":installGraalVmAmd64")
-  configureExecutable(
-    buildInfo.os.isMacOsX,
-    buildInfo.graalVmAmd64,
-    layout.buildDirectory.file("executable/pkl-macos-amd64")
-  )
+  if (buildInfo.os.isMacOsX) {
+    dependsOn(":installGraalVmAmd64")
```

Comment:
Curious if there's a better way to describe a conditional task dependency than this?

Reply:
Why is this dependency conditional?

Reply:
Without the `if`, I am seeing that `installGraalVmAmd64` gets executed on every OS/arch.

Reply:
Sounds like something else is wrong with the build script. The `if` doesn’t make sense to me.

Reply:
Task `buildNative` depends on `macExecutableAmd64`, `macExecutableAarch64`, etc, etc.

I am seeing that these tasks themselves don't run if `enabled = false`, but its dependent tasks still run, unless I _also_ don't call `dependsOn` at all.